src/App.tsx

```diff
@@ -1,93 +1,244 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useState, useMemo,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorType } from './types/ErrorType';
+import {
+  getTodos, createTodo, removeTodo, patchTodo,
+} from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+const getVisibleTodos = (todos: Todo[], filterBy: string) => {
+  switch (filterBy) {
+    case Status.ACTIVE:
+      return todos.filter(item => !item.completed);
+    case Status.COMPLETED:
+      return todos.filter(item => item.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<ErrorType>(ErrorType.NONE);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadedTodoIds, setLoadedTodoIds] = useState([0]);
+
+  const [userID, setUserID] = useState(0);
```

Comment:
check this bug
![chrome-capture-2023-6-6](https://github.com/mate-academy/react_todo-app/assets/33029864/a1eb0c3d-ec27-4d36-b415-d74050dc641c)

after login first time visible todos for user with id 0.
they fetch after logout
for fix it consider prevent fetching todos for user with id 0
and after logout setTodos([]);